libcalico-go/lib/ipam/ipam_block_reader_writer.go

```diff
@@ -319,9 +320,9 @@ func (rw blockReaderWriter) releaseBlockAffinity(ctx context.Context, host strin
 	b := allocationBlock{obj.Value.(*model.AllocationBlock)}
 
 	// Check that the block affinity matches the given affinity.
-	if b.Affinity != nil && !hostAffinityMatches(host, b.AllocationBlock) {
+	if b.Affinity != nil && !hostAffinityMatches(affinityCfg.Host, b.AllocationBlock) {
```

Comment:
`hostAffinityMatches` needs to be aware of `affinityCfg.Type` to make sure we don't release blocks that are of the wrong type. Otherwise there is a risk that we release a block of type `virtual` when we want to release a `host` block (and vice-versa).